src/components/entries/Tooltip.js

```diff
@@ -0,0 +1,123 @@
+import { useRef } from 'preact/hooks';
+import {
+  useEffect,
+  useState
+} from 'react';
+
+/**
+ * @param {Object} props
+ * @param {String} props.labelId
+ * @param {String} props.value
+ */
+export default function Tooltip(props) {
+  const {
+    value,
+    labelId
+  } = props;
+
+  const [ show, setShow ] = useState(false);
+
+  const wrapperRef = useRef(null);
+  const tooltipRef = useRef(null);
+
+  const showTooltip = async () => {
+
+    setTimeout(() => {
+      setShow(true);
+    }, 200);
+  };
+  const hideTooltip = () => setShow(false);
+
+  const isTooltipHovered = ({ x, y }) => {
+    const tooltip = tooltipRef.current;
+    const wrapper = wrapperRef.current;
+
+    return tooltip && (
+      inBounds(x, y, wrapper.getBoundingClientRect()) ||
+      inBounds(x, y, tooltip.getBoundingClientRect())
+    );
+  };
+
+  useEffect(() => {
+    const { current } = wrapperRef;
+
+    if (!current) {
+      return;
+    }
+
+    const hideHoveredTooltip = (e) => {
+      if (!isTooltipHovered({ x: e.x, y: e.y })) {
+        hideTooltip();
+      }
+    };
+
+    const hideFocusedTooltip = (e) => {
+      const { relatedTarget } = e;
+      const isTooltipChild = (el) => el && el.closest('.bio-properties-panel-tooltip');
+
+      if (!isTooltipChild(relatedTarget)) {
+        hideTooltip();
+      }
+    };
+
+    const hideTooltipViaEscape = (e) => {
+      e.code === 'Escape' && hideTooltip();
+    };
+
+    current.addEventListener('mouseover', showTooltip);
+    current.addEventListener('focusin', showTooltip);
+    current.addEventListener('keydown', hideTooltipViaEscape);
+    document.addEventListener('focusout', hideFocusedTooltip);
+    document.addEventListener('mousemove', hideHoveredTooltip);
+
+    return () => {
+      current.removeEventListener('mouseover', showTooltip);
+      current.removeEventListener('focusin', showTooltip);
+      current.removeEventListener('keydown', hideTooltipViaEscape);
+      document.removeEventListener('mousemove', hideHoveredTooltip);
+      document.removeEventListener('focusout', hideFocusedTooltip);
+    };
+  }, [ wrapperRef ]);
+
+  if (value) {
+    return (
+      <div class="bio-properties-panel-tooltip-wrapper" tabIndex="0" ref={ wrapperRef }>
+        {props.children}
+        {show ?
+          <div
+            class="bio-properties-panel-tooltip"
+            role="tooltip"
+            id="bio-properties-panel-tooltip"
+            aria-labelledby={ labelId }
+            style={ getTooltipPosition(wrapperRef.current) }
+            ref={ tooltipRef }
+          >
+            <div class="bio-properties-panel-tooltip-content">
+              {value}
+            </div>
+            <div class="bio-properties-panel-tooltip-arrow" />
+          </div>
+          : null
+        }
+      </div>
+    );
+  } else {
+    return props.children;
```

Comment:
I don't know if this affects the performance a lot, but currently we register all the hooks even for non-tooltipped entries. Perhaps a cleaner implementation could use two elements:

```jsx
export function TooltipWrapper(props) {
  if (!props.tooltip) {
    return props.children;
  }

  return <Tooltip { ...props } />
}

function Tooltip(props) {
  /* there goes implementation */
}
```

Reply:
Good idea